kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/InJarArtifactResolver.java

```diff
@@ -0,0 +1,169 @@
+            if (url.isEmpty()) {
+                return null;
+            }
+            URL pomFile = classLoader.getResource(url.get(0) + "!/META-INF/maven/" + releaseId.getGroupId() + "/" + releaseId.getArtifactId() + "/pom.xml");
+            InputStream pomStream = pomFile.openStream();
+            MavenProject mavenProject = MavenProjectLoader.parseMavenPom(pomStream);
+            PomParser artifactPomParser = new EmbeddedPomParser(mavenProject);
+            pomStream.close();
+            return artifactPomParser;
+        } catch (IOException e) {
+            return null;
+        }
+
+    }
+    private boolean isInJarFolder(String name) {
+        String[] paths = new String[]{"BOOT-INF/classes/KIE-INF/", "KIE-INF/lib/"};
+        for (String path : paths) {
+            if (name.startsWith(path) && name.endsWith(".jar")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public ArtifactLocation resolveArtifactLocation(AFReleaseId releaseId) {
+        Optional<URL> url = tryInJar(releaseId);
+        if (url.isPresent()) {
+            log.info("Resolved in jar repository {}", url);
+            DefaultArtifact artifact = new DefaultArtifact(releaseId.toExternalForm());
+            return new ArtifactLocation(artifact.setFile(new File(url.get().toString())), url.get(), true);
+        }
+        return null;
+
+    }
+
+    @Override
+    public Artifact resolveArtifact(AFReleaseId releaseId) {
+        Optional<URL> url = tryInJar(releaseId);
+        if (url.isPresent()) {
+            log.info("Resolved in jar repository {}", url);
+            DefaultArtifact artifact = new DefaultArtifact(releaseId.toExternalForm());
+            return artifact.setFile(new File(url.get().toString()));
+        }
+        return null;
+    }
+
+    private Optional<URL> tryInJar(String artifactName) {
+        for (URL inJarURL : jarRepository) {
+            if (inJarURL.getFile().endsWith(artifactName)) {
+                return Optional.of(inJarURL);
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<URL> tryInJar(AFReleaseId releaseId) {
+        return tryInJar(toFile(releaseId));
+    }
+
+    private String toFile(AFReleaseId releaseId) {
+        return releaseId.getArtifactId() + "-" + releaseId.getVersion() + ".jar";
+    }
+
+    @Override
+    public List<DependencyDescriptor> getArtifactDependecies(String artifactName) {
+        AFReleaseId releaseId = new AFReleaseIdImpl(artifactName);
+        PomParser pomParser = buildPomParser(releaseId);
+        return pomParser != null ? pomParser.getPomDirectDependencies(DependencyFilter.COMPILE_FILTER) : emptyList();
+    }
+
+    @Override
+    public List<DependencyDescriptor> getPomDirectDependencies(DependencyFilter dependencyFilter) {
+        return pomParser.getPomDirectDependencies(dependencyFilter);
```

Comment:
if pomParser is null, it will throw NPE
```suggestion
        return pomParser != null ? pomParser.getPomDirectDependencies(dependencyFilter) : emptyList();
```

Reply:
done